src/components/[guild]/JoinModal/utils/mapAccessJobState.ts

```diff
@@ -50,14 +50,23 @@ const mapAccessJobState = (progress: JoinJob, isLoading: boolean) => {
       ? (progress as any).position
       : null
 
-  const requirements = progress["children:access-check:jobs"]
+  /**
+   * After the "access-check" step, we can use progress.requirementAccesses for live
+   * data, since that includes the final results (even if the guild has "chained"
+   * roles)
+   */
+  const requirementAccesses =
+    !!progress["completed-queue"] &&
```

Comment:
This was a little bit difficult to read, I'd suggest to move these first two lines to an isAccessCheckComplete variable so it's self explanatory what we're doing here. Other than this, everything else looks good, the comments are great, approved! (I haven't tested but saw you've done a thorough testing in DC so that's great too)

Reply:
Good point, refactored in 565612db513c894ceff0e381005d297cc9a49406!